### PR TITLE
[AMORO-2681] Upgrade Optimizer Flink Docker File

### LINKED ISSUE #2681: [Improvement]: Upgrade Optimizer Flink Docker File
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Upgrade Optimizer Flink Docker File Flink Version And aws version Jars

### How should we improve?

- Upgrade Optimizer Flink Docker File Flink Version to 1.18.1
- aws version jar upgrade to 2.24.12

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by czy006**
## Why are the changes needed?

Close #2681.

## Brief change log

- Upgrade Optimizer Flink Docker File Flink and aws versions

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible
- [x] Add screenshots for manual tests if appropriate
- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by baiyangtx — COMMENTED** [review]
Should we modified the github ci?

**Comment by czy006** [comment]
> Should we change dockerfile of spark-optimizer image to align versions of aws jars?

aws jars versions is pom awssdk version,I think maybe need it

**Comment by czy006** [comment]
> Should we modified the github ci?

[CI](https://github.com/NetEase/amoro/blob/master/.github/workflows/docker-images.yml#L105) lower flink version is 1.14.6.I think CI not need to change

File: docker/optimizer-flink/Dockerfile [hunk 1]
```diff
@@ -14,19 +14,23 @@
...
 ARG OPTIMIZER_JOB=optimizer-job.jar
+ARG AWS_VERSION=2.24.12
 
-RUN cd $FLINK_HOME/lib \
-    && wget ${MAVEN_MIRROR}/software/amazon/awssdk/bundle/2.20.5/bundle-2.20.5.jar \
-    && wget ${MAVEN_MIRROR}/software/amazon/awssdk/url-connection-client/2.20.5/url-connection-client-2.20.5.jar \
-    && wget ${MAVEN_MIRROR}/org/apache/flink/flink-shaded-hadoop-2-uber/2.8.3-10.0/flink-shaded-hadoop-2-uber-2.8.3-10.0.jar \
-    && chown flink:flink *.jar \
-    && mkdir -p $FLINK_HOME/usrlib
+WORKDIR $FLINK_HOME/lib
```
Comment (tcodehuber) [inline docker/optimizer-flink/Dockerfile:25]: I suggest using `RUN cd $FLINK_HOME/lib \`  instead of docker instruction `WORKDIR`. Furthermore, we can keep consistent code style with spark optimizer's dockerfile.

File: docker/optimizer-flink/Dockerfile [hunk 2]
```diff
@@ -14,19 +14,23 @@
...
-ADD $OPTIMIZER_JOB $FLINK_HOME/usrlib/optimizer-job.jar
+RUN wget ${MAVEN_MIRROR}/software/amazon/awssdk/bundle/${AWS_VERSION}/bundle-${AWS_VERSION}.jar \
+ && wget ${MAVEN_MIRROR}/software/amazon/awssdk/url-connection-client/${AWS_VERSION}/url-connection-client-${AWS_VERSION}.jar \
+ && wget ${MAVEN_MIRROR}/org/apache/flink/flink-shaded-hadoop-2-uber/2.8.3-10.0/flink-shaded-hadoop-2-uber-2.8.3-10.0.jar \
+ && chown flink:flink *.jar \
+ && mkdir -p $FLINK_HOME/usrlib
+
+COPY $OPTIMIZER_JOB $FLINK_HOME/usrlib/optimizer-job.jar
+
+WORKDIR /
```
Comment (tcodehuber) [inline docker/optimizer-flink/Dockerfile:35]: Can be removed

File: docker/optimizer-flink/Dockerfile [hunk 3]
```diff
@@ -14,19 +14,23 @@
 #  See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG FLINK_VERSION=1.14
+ARG FLINK_VERSION=1.18.1-java8
```
Comment (tcodehuber) [inline docker/optimizer-flink/Dockerfile:17]: default flink_version in build.sh is 1.15.3, which uses java11. So ARG FLINK_VERSION=1.18.1 is needed?

Comment (czy006) [reply docker/optimizer-flink/Dockerfile:17]: we should direct users to default to the latest supported version of Flink

**Review by tcodehuber — APPROVED** [review]
LGTM
